source/best-practices/backup.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+#
+# 使用 rsync 命令备份指定目录
+#
+#        作者： 姚家园、田冬冬
+# 最近修改日期： 2021年2月2日
+#
+
+source="${HOME}/workspace/source"  # 源目录
+backup="${HOME}/workspace/backup"  # 备份目录（通常为移动硬盘的挂载目录）
+log="${backup}/backup.log"         # 备份目录下的备份日志
+
+# 检查源目录
+if [ ! -d "${source}" ]; then                 # 源目录不存在，退出程序
+    echo "[${source}] does not exists!"
+    exit
+elif [[ `ls ${source} | wc -w` =~ 0 ]]; then  # 源目录是空目录，退出程序
+    echo "[${source}] is empty!"
+    exit
+fi
```

Comment:
```suggestion
elif [[ $(ls ${source} | wc -w) =~ 0 ]]; then  # 源目录是空目录，退出程序
    echo "[${source}] is empty!"
    exit
fi
```

https://stackoverflow.com/questions/9449778/what-is-the-benefit-of-using-instead-of-backticks-in-shell-scripts/9449793

Reply:
Remove `-A`. Otherwise, the hidden files and dirs will be counted.

Reply:
ls -A List all entries except for . and ..

I think it means -A is required, no?

Reply:
Add `"` for `$source` to deal with possible whitespace in the folder path.

Reply:
> ls -A List all entries except for . and ..
> 
> I think it means -A is required, no?

We usually ignore hidden files and dirs, right?

Reply:
I think this is not a big problem for $HOME, because it is empty usually.

It is a problem for a folder with only hidden dirs and files because `dirs=( ${source}/* )` only work for non-hidden dirs.

Reply:
I see what you mean.